packages/transactions/src/signer.ts

```diff
@@ -19,6 +19,26 @@ export class TransactionSigner {
     this.originDone = false;
     this.checkOversign = true;
     this.checkOverlap = true;
+
+    // If multi-sig spending condition exists, iterate over
+    // auth fields and reconstruct sigHash
+    let spendingCondition = transaction.auth.spendingCondition;
+    if (spendingCondition && !isSingleSig(spendingCondition)) {
+      spendingCondition.fields.forEach(field => {
```

Comment:
Should this have a similar check to the other signing function? like 
```ts
      if (
        spendingCondition.fields.length >= spendingCondition.signaturesRequired
      ) {
        throw new Error('Origin would have too many signatures');
      }
```

Reply:
Good idea! I wonder if perhaps this check should happen instead (or also) in the `deserializeMultiSigSpendingCondition` function...

Reply:
Why is it `>=`? is the `==` case also an error? Can we test one of these error cases?